### PR TITLE
daemon: also check file mode when comparing

### PR DESCRIPTION
**Pull request by jlebon**
Otherwise we'll think we successfully reconciled even if the modes
differ.

### CODE REVIEW

File: pkg/daemon/update.go [hunk 1]
```diff
@@ -19,6 +19,11 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+const (
+	DefaultDirectoryPermissions os.FileMode = 0755
```
Comment (sdemos) [inline pkg/daemon/update.go]: it doesn't look like this is used anywhere?

File: pkg/daemon/daemon.go [hunk 2]
```diff
@@ -322,7 +327,16 @@ func (dn *Daemon) checkFiles(files []ignv2_2types.File) bool {
 // checkFileContents reads the file from the filepath and compares its contents
 // with the  expectedContent. It logs an error in case of an error or mismatch
 // and returns the status of the evaluation.
-func checkFileContents(filePath, expectedContent string) bool {
+func checkFileContentsAndMode(filePath, expectedContent string, mode os.FileMode) bool {
+	fi, err := os.Lstat(filePath)
+	if err != nil {
+		glog.Errorf("could not read file: %q, error: %v", filePath, err)
```
Comment (sdemos) [inline pkg/daemon/daemon.go]: might be better if it's `could not stat file` or `could not get file info` or something like that.

File: pkg/daemon/daemon.go [hunk 3]
```diff
@@ -322,7 +327,16 @@ func (dn *Daemon) checkFiles(files []ignv2_2types.File) bool {
 // checkFileContents reads the file from the filepath and compares its contents
```
Comment (sdemos) [inline pkg/daemon/daemon.go]: should update doc comment with new name and additional info on file mode handling

File: pkg/daemon/update.go (hunk 1 above)
Comment (jlebon) [reply pkg/daemon/update.go]: It does now! :) I added a commit to make sure we match Ignition there.

**Comment by sdemos** [comment]
woah, wrecked.

/retest

**Pull request merged**